File: src/main/java/me/iankit/simplechat/chat/Group.java
package me.iankit.simplechat.chat;

import lombok.Getter;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

@Getter
@Document(collection = "group")
public class Group {
    private final String groupName;
    private final String createdBy;
    @Id
    private String id;

    public Group(String groupName, String createdBy) {
        this.groupName = groupName;
        this.createdBy = createdBy;
    }
}

File: src/main/java/me/iankit/simplechat/chat/Group.kt
package me.iankit.simplechat.chat

import org.springframework.data.annotation.Id
import org.springframework.data.mongodb.core.mapping.Document

@Document(collection = "group")
class Group(val groupName: String, val createdBy: String) {
    @get:Id
    var id: String? = null
        protected set

}